lib/util/utils.ts

```diff
@@ -28,6 +28,11 @@ export function getFetch(): Function {
  */
 export const myFetch = getFetch();
 
+/**
+ * A constant that indicates whether the environment is node.js or browser based.
+ */
+export const isNode = typeof navigator === 'undefined' && typeof process !== 'undefined'
```

Comment:
@geocachingsyntaxerror - This is the logic that I found from a package named [detect-browser](https://github.com/DamonOehlman/detect-browser/blob/master/index.js#L69) which is used to detect whether the app is in a node.js environment or not. I did a basic test in the node REPL and loading one of the samples from this repo in the browser. It worked as expected in both the environments. 

Can you please pull the changes from detect branch of my fork of ms-rest-js and let me know if this works for react-native as well?

Reply:
FYI, If this change does not work then we can try one more package [is-in-browser](https://www.npmjs.com/package/is-in-browser).

Reply:
I pulled your detect branch and it worked as expected in react-native; isNode returns false as expected. Thank you.
<img width="1461" alt="detect" src="https://user-images.githubusercontent.com/8315604/35318213-2fff3748-008f-11e8-9e9a-7c9f722296cc.png">